test/wasm_components_test.exs

```diff
@@ -0,0 +1,43 @@
+defmodule Wasmex.WasmComponentsTest do
+  use ExUnit.Case, async: true
+
+  alias Wasmex.Engine
+  alias Wasmex.EngineConfig
+
+  test "invoke component func" do
```

Comment:
regarding test setup, I'd love to see one (bigger) wasm component that tests all (or at least most) input/output variations with all types. Something similar to what wasmtime-ruby does or what my PR does in the test setup.

However, I love your descriptive examples. Maybe we can choose *one* "good" example to show-off component support in a test. what do you think? Feel free to steal the test setup in my MR if you want.

Reply:
I added a component types test by stealing more from wasmtime-rb :) Their example was built in rust not javascript so it is waaay faster to boot. I may or may not keep the original around. It probably makes sense to move it out of a test into an example directory or something. Because it was implemented in js it will always be super slow to load

Reply:
sounds good. I really like the approach wasmtime-rb did in their testing. very systematic and easy to see what's supported.

regarding an example: I'm happy to have one more wasm_component that is a tiny bit more complex (probably something using wasi, which the wasmtime-rb test does not use to my knowledge)

All other examples, IMO should live in separate repositories. Reason is that they should either:

1. live in this repo (in an example repo), but then we need to test them so they don't get stale
2. or live in example repository/repositories - which can test on their own and are way better "templates" is someone wants to start a project

To me 2. looks like having more benefits with less disadvantages for the main repo

Reply:
Yeah, i think moving the more complex components into separate examples repos is the way to go. This will keep are tests faster as well.